### PR TITLE
Add docs on cluster configuration

### PR DESCRIPTION
**Pull request by yuvipanda**

### CODE REVIEW

**Comment by yuvipanda** [comment]
@ryanlovett you might have bits here that I've missed.

Partially based off document built in collaboration with @zoeyl for setting up spring 2019 cluster.

/cc @felder 

Might be of interest to @consideratio in regards to https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/1069

File: docs/topic/cluster-config.rst [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+We prefer using a managed Kubernetes service (such as `Google Kubernetes Engine
+<https://cloud.google.com/kubernetes-engine/>`_). This document lays out our
+preferred cluster configuration on various cloud providers.
+
+Google Kubernetes Engine
+========================
+
+In our experience, Google Kubernetes Engine (GKE) has been the most stable,
+performant & reliable managed kubernetes service. We prefer running on this
```
Comment (ryanlovett) [inline docs/topic/cluster-config.rst]: performant, and reliable

File: docs/topic/cluster-config.rst [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        --enable-autoscaling \
+        --max-nodes=20 --min-nodes=2 \
+        --region=us-central1 --node-locations=us-central1-b \
+        --image-type=ubuntu \
+        --disk-size=100 --disk-type=pd-standard \
+        --machine-type=n1-highmem-8 \
+        --cluster-version latest \
+        <cluster-name>
+
+We will try explain the various arguments to this
```
Comment (ryanlovett) [inline docs/topic/cluster-config.rst]: .

File: docs/topic/cluster-config.rst [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+By default, asking for highly available masters also asks for 3x the node count,
+spread across multiple zones. We don't want that, since all our user pods have
+in-memory state & can't be relocated. Specifying ``--node-locations`` explicitly
+lets us control how many and which zones the nodes are located in.
+
+Region / Zone selection
+-----------------------
+
+We generally use the ``us-central1`` region and a zone in it for our clusters -
```
Comment (ryanlovett) [inline docs/topic/cluster-config.rst]: Is it worth anticipating latency questions here? I don't think it has mattered for us.

File: docs/topic/cluster-config.rst [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+Ubuntu as our `node operating system <https://cloud.google.com/kubernetes-engine/docs/concepts/node-images>`_.
+The default (Container Optimized OS) does not have NFS support enabled.
+
+Disk Size
+---------
+
+``--disk-size`` sets the size of the root disk on all the kubernetes nodes. This
+isn't used for any persistent storage such as user home directories. It is only
+used ephemerally for the operations of the cluster - primarily storing docker
+images & other temporary storage. We can make this larger if we use a large number
```
Comment (ryanlovett) [inline docs/topic/cluster-config.rst]: s/&/and/

**Comment by ryanlovett** [comment]
@yuvipanda Looks good! Made very minor comments.

**Comment by yuvipanda** [comment]
@ryanlovett thanks! '&' and 'and' are equivalent no?

**Comment by yuvipanda** [comment]
@ryanlovett I generally defer to whoever has stronger opinions - and I appreciate that you do!

I've updated with these changes. In the future, can you try using https://github.blog/changelog/2018-10-16-suggested-changes/ for such typographical changes?

**Comment by ryanlovett** [comment]
@yuvipanda Will do. Btw,, I **strongly** feel you should give me your headphones.

**Comment by consideRatio** [comment]
@yuvipanda thank you for pinging me I do indeed appreciate reading this!!

File: docs/topic/cluster-config.rst [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+#. Students generally are memory limited than CPU limited. In fact, while we
+   have a hard limit on memory use per-user pod, we do not have a CPU limit -
+   it hasn't proven necessary.
+
+#. We try overprovision clusters by about 2x - so we try to fit about 100G of total RAM
+   use in a node with about 50G of RAM. This is accomplished by setting the memory
+   request to be about half of the memory limit on user pods. This leads to massive
+   cost savings, and works out ok.
+
+#. There is a kubernetes limit on 100 pods per node.
```
Comment (consideRatio) [inline docs/topic/cluster-config.rst]: I think this is somewhat outdated, and that it is now "about 100" as I recall it is closer to 110.

**Pull request merged**